src/clonemap.rs

```diff
@@ -0,0 +1,859 @@
+}
+
+impl<K, V> CloneConMap<K, V>
+where
+    K: Clone + Hash + Eq + 'static,
+    V: Clone + 'static,
+{
+    /// Creates a new empty map.
+    pub fn new() -> Self {
+        Self::with_hasher(RandomState::default())
+    }
+}
+
+impl<K, V, S> CloneConMap<K, V, S>
+where
+    K: Clone + Hash + Eq + 'static,
+    V: Clone + 'static,
+    S: BuildHasher,
+{
+    /// Inserts a new element.
+    ///
+    /// Any previous element with the same key is replaced and returned.
+    pub fn insert(&self, key: K, value: V) -> Option<Element<K, V>> {
+        self.insert_element(Element::new(key.clone(), value.clone()))
+    }
+
+    /// Looks up or inserts an element.
+    ///
+    /// It looks up an element. If it isn't present, the provided one is
+    /// inserted instead. Either way, an element is returned.
+    pub fn get_or_insert(&self, key: K, value: V) -> ExistingOrNew<Element<K, V>> {
+        self.get_or_insert_with(key, || value)
+    }
+
+    /// Looks up or inserts a newly created element.
+    ///
+    /// It looks up an element. If it isn't present, the provided
+    /// closure is used to create a new one insert it. Either way, an
+    /// element is returned.
+    ///
+    /// # Quirks
+    ///
+    /// Due to races in case of concurrent accesses, the closure may be
+    /// called even if the value is not subsequently inserted and an
+    /// existing element is returned. This should be relatively rare
+    /// (another thread must insert the new element between this method
+    /// observes an empty slot and manages to insert the new element).
+    pub fn get_or_insert_with<F>(&self, key: K, create: F) -> ExistingOrNew<Element<K, V>>
+    where
+        F: FnOnce() -> V,
+    {
+        self.get_or_insert_with_element(key, |key| {
+            let value: V = create();
+            Element::new(key.clone(), value.clone())
+        })
+    }
+
+    /// Looks up or inserts a default value of an element.
+    ///
+    /// This is like [get_or_insert_with][CloneConMap::get_or_insert_with],
+    /// but a default value is used instead of manually providing a
+    /// closure.
+    pub fn get_or_insert_default(&self, key: K) -> ExistingOrNew<Element<K, V>>
+    where
+        V: Default,
+    {
+        self.get_or_insert_with(key.clone(), V::default)
+    }
+}
+
+impl<K, V, S> CloneConMap<K, V, S>
+where
+    K: Clone + Hash + Eq,
+    V: Clone,
+    S: BuildHasher,
+{
+    /// Creates a new empty map, but with the provided hasher implementation.
+    pub fn with_hasher(hasher: S) -> Self {
+        Self {
+            raw: Raw::with_hasher(hasher),
+        }
+    }
+
+    /// Inserts a new element.
+    ///
+    /// This acts the same as [insert][CloneConMap::insert], but takes
+    /// the already created element. It can be used when:
+    ///
+    /// * `V: ?Sized`.
+    /// * You want to insert the same element into multiple maps.
+    pub fn insert_element(&self, element: Element<K, V>) -> Option<Element<K, V>> {
```

Comment:
As described, I believe there's no practical advantage for all the `_element` methods. The code can probably be just moved inside the normal ones and get rid of them.